### PR TITLE
Add TypeScript template to project wizard

### PR DESCRIPTION
**Pull request by rli**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsSamProjectWizard.kt [hunk 1]
```diff
@@ -67,3 +68,17 @@ class SamHelloWorldNodeJs : SamAppTemplateBased() {
 
     override val dependencyManager: String = "npm"
 }
+
+class SamHelloWorldNodeJsTypeScript : SamAppTemplateBased() {
+    override fun displayName() = message("sam.init.template.hello_world_typescript.name")
+
+    override fun description() = message("sam.init.template.hello_world_typescript.description")
+
+    override fun supportedZipRuntimes(): Set<LambdaRuntime> = setOf(LambdaRuntime.NODEJS12_X)
```
Comment (abrooksv) [inline jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsSamProjectWizard.kt:77]: no NODEJS14_X, NODEJS10_X?

File: resources/resources/software/aws/toolkits/resources/MessagesBundle.properties [hunk 2]
```diff
@@ -696,6 +696,8 @@ sam.init.template.hello_world.description=A basic SAM app
 sam.init.template.hello_world.name=AWS SAM Hello World
 sam.init.template.hello_world_gradle.name=AWS SAM Hello World (Gradle)
 sam.init.template.hello_world_maven.name=AWS SAM Hello World (Maven)
+sam.init.template.hello_world_typescript.description=A sample app using TypeScript
```
Comment (manodnyab) [inline resources/resources/software/aws/toolkits/resources/MessagesBundle.properties:699]: Should we specify in the description that it's a 'sample Hello World app'?

Comment (rli) [reply resources/resources/software/aws/toolkits/resources/MessagesBundle.properties:701]: it's more complicated than the standard hello world, which is why the description is more generic

**Pull request merged**